src/main/kotlin/lotto/domain/LottoPurchaser.kt

```diff
@@ -0,0 +1,18 @@
+package lotto.domain
+
+/**
+ * 입력 받은 금액 만큼의 로또를 돌려준다.
+ */
+class LottoPurchaser(
+    private val lottoGenerator: LottoGenerator
+) {
+
+    fun purchase(inputAmount: Int): List<Lotto> {
+        val numOfLotto = getPurchasableNum(inputAmount)
+        return lottoGenerator.generate(numOfLotto)
+    }
+
+    fun getPurchasableNum(inputAmount: Int): Int {
+        return inputAmount / Lotto.PRICE
+    }
+}
```

Comment:
`로또 구매자`라는 새로운 도메인을 설계하셨네요!
보통 구매자는 판매자에게 물건을 구입하는만큼, `LottoGenerator`에게 로또를 구매하는 것이 아니라 `로또 판매자`에게 로또를 구매하도록 바꿔보면 어떨까요? 그리고 구매한 로또를 이 객체에서 관리해봐도 좋을 것 같아요🙂

```suggestion
class LottoPurchaser(
    private val lottoSeller: LottoSeller
    lottos: List<Lotto> = emptyList()
) {

    private val purchasedLottos: MutableList<Lotto> = lottos.toMutableList()

    fun purchase(inputAmount: Int) {
        val purchasedLottos = lottoSeller.sell(inputAmount)
        return purchasedLottos.addAll(purchasedLottos)
    }
}

```